### DISCUSSION TITLE
FirefoxPWA 1.0.0

### DISCUSSION THREAD

**Discussion by filips123** [filips123/PWAsForFirefox#45] [Announcements]
## Important

* Original Homebrew formula had a bug that prevented PWAs from launching on macOS. I reuploaded the formula with a fix. To install the fixed version, run:

  ```bash
  brew update # To update all taps
  brew reinstall firefoxpwa # To reinstall FirefoxPWA with a fix
  ```

  Then, edit/update all existing PWAs through extension or command-line program to apply the fix to PWAs. Alternatively, you can remove and reinstall your PWAs.

* ~~Some antivirus programs (Microsoft Defender) incorrectly flagged the 32-bit Windows version as malicious. I will try to resolve this in the future. Prefer using the 64-bit version if possible.~~ This has been resolved. Update your malware definitions in case the program is still detected as malware.

## News

This is the first "stable" release of the FirefoxPWA project! 🎉

This means most features that I initially planned are implemented, and the project is ready for wider use and adoption. It does not mean the project is completely finished and bug-free; There are still some [issues](https://github.com/filips123/FirefoxPWA/issues) and [limitations](https://github.com/filips123/FirefoxPWA#current-limitations) that I would like to fix eventually, but they will probably require more time and work.

Starting with this release, FirefoxPWA has been officially packaged for various package managers, such as Windows Package Manager, Chocolatey, Homebrew and others, as well as packagecloud.io-hosted repository for DEB and RPM packages. It is not required to use them, but they provide support for easier updating and installing.

If you want to package FirefoxPWA for your distribution or package manager, or have already packaged it, please let me know, so I can help with the packaging and list it in the installation instructions. If the packaging platform supports any auto-submission/uploading feature, please also let me know, as I might integrate it directly with GitHub Actions to make sure the packages are always up-to-date.

Feel free to also check the project [on ProductHunt](https://www.producthunt.com/posts/firefoxpwa) and leave a review there.

## Added

* **Support to disable autohide for mute widget:**
  You can now disable autohiding for the mute widget to display it even if nothing is playing. You can do this in the Firefox customize page, similarly to the normal downloads widget.

* **Support to generate missing icons from the site/shortcut names:**
  If the site does not provide any valid icons, the icon will now be generated from the first letter of the site name. The same applies to site shortcuts. If you already have sites without icons, you need to update/edit them to apply the generated icon.

* **Support to specify runtime arguments (#39):**
  You can now specify arguments that are passed to the Firefox instance used by FirefoxPWA. To specify arguments for all sites, add the `arguments` key to the FirefoxPWA JSON config file with a list of arguments. To specify arguments when launching the site from the command line, provide them as console arguments after `--`. Check [this comment](https://github.com/filips123/FirefoxPWA/issues/39#issuecomment-894829747) for more info.

* **Complete integration for macOS (#33):**
  Installed sites on macOS will now properly be added to the macOS applications list, and have their own window with an icon. If you already have sites on macOS, you need to update/edit them to apply the system integration. However, there are still [some limitations and problems](https://github.com/filips123/FirefoxPWA#current-limitations) with system integration, so keep that in mind when using PWAs.

* **More distribution options and auto publishing:**
  You can now install and update FirefoxPWA through various package managers and other distribution options. All supported options are listed in the extension install/update page, and in the [README](https://github.com/filips123/FirefoxPWA/tree/main/native#installation).

## Changed

* Make popup card and footer sticky.
* Improve dark mode for setup pages.
* Reorganize XDG and Apple categories.

## Fixed

* Broken browser widgets are now fixed.
* Close widget now works in multiple windows.
* Xfce now displays the correct PWA icon.
* Fresh profiles no longer overwrite initial page and have misplaced widgets.

<hr /><em>This discussion was created from the release <a href='https://github.com/filips123/FirefoxPWA/releases/tag/v1.0.0'>FirefoxPWA 1.0.0</a>.</em>

**Comment by scgf** [comment]
How do I download the macOS release? All I can see are releases for Linux and Windows. I just tried Homebrew and it claims not to know anything about FirefoxPWA.

**Reply by filips123** [reply]
You need to install it from my [custom tap](https://github.com/filips123/homebrew-firefoxpwa). However, because my auto-deployment script broke, 1.0.0 is not yet available. I will manually add it soon.

**Reply by filips123** [reply]
Version 1.0.0 should now be available in my tap. To install it, run:

```bash
brew tap filips123/firefoxpwa # Activate FirefoxPWA tap
brew install firefoxpwa # Install FirefoxPWA formula
```

If you already added tap before, you will probably need to run `brew update` before installing/upgrading formula.

**Comment by black7375** [comment]
Awesome job.
I always appreciate this project.

**Comment by scgf** [comment]
I've installed the app in Homebrew as described above, then I installed the Firefox PWA extension and installed my first PWA. Unfortunately it doesn't run. I get this error when I try and run "loader" directly:

/Users/phillip/Applications/Log\ in\ to\ Royal\ Bank\ of\ Scotland\ Digital\ Banking.app/Contents/MacOS/loader ; exit;
/Users/phillip/Applications/Log in to Royal Bank of Scotland Digital Banking.app/Contents/MacOS/loader: line 3: /usr/local/bin/firefoxpwa: No such file or directory

**Reply by filips123** [reply]
I reuploaded 1.0.0 to my Homebrew tap with a fix. To install it, run:

```bash
brew update # To update all taps
brew reinstall firefoxpwa # To reinstall FirefoxPWA with a fix
```

Then, update/edit your existing PWAs using the extension to apply the fix, or just delete and re-add them. Please let me know if this solves the issue.

**Reply by scgf** [reply]
Yes, I can confirm the PWAs now work. Thank you.
